### PR TITLE
PRM-446 Added for bereavement reason 'when did the person die' page

### PR DESCRIPTION
**Pull request by ruthdas03**

### CODE REVIEW

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Looks good, just a few changes. AT's fine.

File: conf/app.routes [hunk 1]
```diff
@@ -19,6 +19,16 @@ GET        /sign-out                        controllers.SignOutController.signOu
...
+GET        /when-did-the-person-die/change              controllers.BereavementReasonController.onPageLoadForWhenThePersonDied(mode: Mode = CheckMode)
+POST       /when-did-the-person-die                     controllers.BereavementReasonController.onSubmitForWhenThePersonDied(mode: Mode = NormalMode)
+POST       /when-did-the-person-die/change              controllers.BereavementReasonController.onSubmitForWhenThePersonDied(mode: Mode = CheckMode)
+GET        /when-did-the-person-die                     controllers.BereavementReasonController.onPageLoadForWhenThePersonDied(mode: Mode = NormalMode)
+GET        /when-did-the-person-die/change              controllers.BereavementReasonController.onPageLoadForWhenThePersonDied(mode: Mode = CheckMode)
+POST       /when-did-the-person-die                     controllers.BereavementReasonController.onSubmitForWhenThePersonDied(mode: Mode = NormalMode)
+POST       /when-did-the-person-die/change              controllers.BereavementReasonController.onSubmitForWhenThePersonDied(mode: Mode = CheckMode)
```
[the remark this answers is not in the document]
Comment (ruthdas03) [reply conf/app.routes]: Removed duplicate routes.

File: it/controllers/BereavementReasonControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+      request.header.status shouldBe INTERNAL_SERVER_ERROR
+    }
+
+    "return 303 (SEE_OTHER) when the user is not authorised" in {
+      AuthStub.unauthorised()
+      val request = await(buildClientForRequestToApp(uri = "/when-did-the-person-die").get())
+      request.status shouldBe SEE_OTHER
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (ruthdas03) [reply it/controllers/BereavementReasonControllerISpec.scala:70]: Added /POST for /when-did-the-person-die and in the POST added the SEE_OTHER with Location "#"

File: test/controllers/BereavementReasonControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+                | "date.day": 1,
+                | "date.month": 2,
+                | "date.year": 2021
+                |}
+                |""".stripMargin))))
+          status(result) shouldBe SEE_OTHER
+          redirectLocation(result).get shouldBe "#"
+          await(result).session.get(SessionKeys.whenDidThePersonDie).get shouldBe LocalDate.of(2021, 2, 1).toString
+        }
+
```
[the remark this answers is not in the document]
Comment (ruthdas03) [reply test/controllers/BereavementReasonControllerSpec.scala:109]: Added the above change for MakingALateAppeal route and Location is "#" (for now)

**Comment by LewisMitchell** [comment]
image

**Review by LewisMitchell — APPROVED** [review]
Looks good to me. AT's pass with rebase

**Pull request merged**